x-pack/dockerlogbeat/pipelinemanager/clientLogReader.go

```diff
@@ -114,6 +117,13 @@ func (cl *ClientLogger) publishLoop(reader chan logdriver.LogEntry) {
 		cl.localLog.Log(constructLogSpoolMsg(entry))
 		line := strings.TrimSpace(string(entry.Line))
 
+		splitName := strings.Split(cl.ContainerMeta.ContainerImageName, ":")
+		containerImageName := splitName[0]
+		containerImageTag := ""
+		if len(splitName) > 1 {
+			containerImageTag = splitName[1]
+		}
```

Comment:
consider to use `strings.IndexRune`, so we don't need to alloc and don't need to panic if the input string is empty (yeah, unlikely). e.g.

```
if idx := strings.IndexRune(cl.ContainerMeta.ContainerImageName, ':'); idx >= 0 {
  ...
} else {
}
```

If we need these fields for every event to be published, better do the split once at init time.